### PR TITLE
configs: Validate dynamic xds config

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: configs: Validate dynamic xds config
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
i noticed that one of the bootstrap files wasnt being tested when it could be - this just makes sure its included

File: docs/BUILD [hunk 1]
```diff
@@ -16,7 +16,9 @@ filegroup(
             "root/**/*.yaml",
         ],
         exclude = [
-            "root/**/envoy-dynamic*.yaml",
+            "root/**/envoy-dynamic-cds-demo.yaml",
```
Comment (dmitri-d) [inline docs/BUILD:19]: Not sure I understand: these expand the exclude list and the files won't be present in the produced file list, while the PR comment says that you wanted to validate them?

Comment (phlax) [reply docs/BUILD:19]: the original glob excluded all of the `envoy-dynamic*.yaml` altho one that matched was bootstrap and should be validated

Comment (dmitri-d) [reply docs/BUILD:19]: ahh, I see now.

**Comment by dmitri-d** [comment]
lgtm

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**